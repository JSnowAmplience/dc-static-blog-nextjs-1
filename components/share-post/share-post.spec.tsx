import React from 'react';
import renderer from 'react-test-renderer';
import SharePost from './share-post';

describe('SharePost', (): void => {
  it('should render using the process.env.URL is defined', (): void => {
    process.env.URL = 'http://example.com/';
    const twitterText = 'Testing Title';
    const wrapper = renderer.create(<SharePost twitterText={twitterText} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when process.env.URL is undefined', (): void => {
    delete process.env.URL;
    const twitterText = 'Testing Title';
    const wrapper = renderer.create(<SharePost twitterText={twitterText} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
