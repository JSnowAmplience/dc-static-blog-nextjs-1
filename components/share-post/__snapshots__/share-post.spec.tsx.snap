// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharePost should render using the process.env.URL is defined 1`] = `
<section
  className="jsx-3818356980"
>
  <h3
    className="jsx-3818356980"
  >
    Share the post
  </h3>
  <div
    className="jsx-3818356980"
  >
    <div
      className="jsx-3818356980"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<script async src=\\"https://platform.linkedin.com/in.js\\" type=\\"text/javascript\\">lang: en_US</script><script type=\\"IN/Share\\" data-url=http://example.com/mocked-path></script>",
        }
      }
      suppressHydrationWarning={true}
    />
    <div
      className="jsx-3818356980"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"https://twitter.com/share?ref_src=twsrc%5Etfw\\" class=\\"twitter-share-button\\" data-text=Testing Title data-show-count=\\"false\\">Tweet</a><script async src=\\"https://platform.twitter.com/widgets.js\\" charSet=\\"utf-8\\"></script>",
        }
      }
      suppressHydrationWarning={true}
    />
  </div>
</section>
`;

exports[`SharePost should render when process.env.URL is undefined 1`] = `
<section
  className="jsx-3818356980"
>
  <h3
    className="jsx-3818356980"
  >
    Share the post
  </h3>
  <div
    className="jsx-3818356980"
  >
    <div
      className="jsx-3818356980"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<script async src=\\"https://platform.linkedin.com/in.js\\" type=\\"text/javascript\\">lang: en_US</script><script type=\\"IN/Share\\" data-url=mocked-path></script>",
        }
      }
      suppressHydrationWarning={true}
    />
    <div
      className="jsx-3818356980"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"https://twitter.com/share?ref_src=twsrc%5Etfw\\" class=\\"twitter-share-button\\" data-text=Testing Title data-show-count=\\"false\\">Tweet</a><script async src=\\"https://platform.twitter.com/widgets.js\\" charSet=\\"utf-8\\"></script>",
        }
      }
      suppressHydrationWarning={true}
    />
  </div>
</section>
`;
